agents/agents-features/agents-features-tokenizer/src/jvmTest/kotlin/ai/koog/agents/features/tokenizer/feature/MessageTokenizerTest.kt

```diff
@@ -0,0 +1,212 @@
+package ai.koog.agents.features.tokenizer.feature
+
+import ai.koog.agents.core.agent.AIAgent
+import ai.koog.agents.core.agent.config.AIAgentConfig
+import ai.koog.agents.core.dsl.builder.forwardTo
+import ai.koog.agents.core.dsl.builder.strategy
+import ai.koog.agents.core.dsl.extension.*
+import ai.koog.agents.core.tools.ToolRegistry
+import ai.koog.agents.testing.feature.withTesting
+import ai.koog.agents.testing.tools.getMockExecutor
+import ai.koog.agents.testing.tools.mockLLMAnswer
+import ai.koog.prompt.dsl.Prompt
+import ai.koog.prompt.dsl.prompt
+import ai.koog.prompt.llm.OllamaModels
+import ai.koog.prompt.message.Message
+import ai.koog.prompt.tokenizer.Tokenizer
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+/**
+ * Test for the MessageTokenizer feature.
+ *
+ * This test verifies that the MessageTokenizer correctly tracks token usage.
+ */
+class MessageTokenizerTest {
+
+    /**
+     * A mock tokenizer that tracks the total tokens counted.
+     *
+     * This implementation counts tokens by simply counting characters and dividing by 4,
+     * which is a very rough approximation but sufficient for testing purposes.
+     * It also keeps track of the total tokens counted across all calls.
+     */
+    class MockTokenizer : Tokenizer {
+        private var _totalTokens = 0
+
+        /**
+         * The total number of tokens counted across all calls to countTokens.
+         */
+        val totalTokens: Int
+            get() = _totalTokens
+
+        /**
+         * Counts tokens by simply counting characters and dividing by 4.
+         * Also adds to the running total of tokens counted.
+         *
+         * @param text The text to tokenize
+         * @return The estimated number of tokens in the text
+         */
+        override fun countTokens(text: String): Int {
+            // Simple approximation: 1 token ≈ 4 characters
+            println("countTokens: $text")
+            val tokens = (text.length / 4) + 1
+            _totalTokens += tokens
+            return tokens
+        }
+
+        /**
+         * Resets the total tokens counter to 0.
+         */
+        fun reset() {
+            _totalTokens = 0
+        }
+    }
+
+    /**
+     * A simple implementation of PromptTokenizer for testing.
+     * This is similar to the OnDemandTokenizer in the main code.
+     */
+    class TestPromptTokenizer(private val tokenizer: Tokenizer) {
+        /**
+         * Counts tokens for a single message.
+         */
+        fun tokensFor(message: Message): Int = tokenizer.countTokens(message.content)
+
+        /**
+         * Counts total tokens for all messages in a prompt.
+         */
+        fun totalTokensSpent(prompt: Prompt): Int = prompt.messages.sumOf { tokensFor(it) }
+    }
+
+    @Test
+    fun testTokenizerFeature() = runTest {
```

Comment:
I don't quite see the purpose of this test. It just tests mock implementations, which is not that useful. We don't want to test mocks in our tests, we want to use these mocks to test some actual behavior.